metadata/metadata-modeshape/src/main/java/com/thinkbiganalytics/metadata/modeshape/BaseJcrProvider.java

```diff
@@ -333,23 +338,70 @@ public T findFirst(String query) {
         }
     }
 
+    protected String getFindAllFilter() {
+        return getFindAllFilter(getEntityQueryStartingPath());
+    }
+
+    protected String getFindAllFilter(String startingPath) {
+        int depth = 0;
+        if (startingPath != null) {
+            depth = startingPath.split("/").length;
+        }
+        return getFindAllFilter(startingPath, depth);
+    }
+
+    protected String getFindAllFilter(String startingPath, int depth) {
+        boolean skip = false;
+        if (skip) {
+            return null;
+        }
+        if (startingPath == null) {
+            return null;
+        } else {
+            return "ISDESCENDANTNODE('" + startingPath + "') AND ( DEPTH() = CAST(" + depth + " AS LONG) ) "; // OR DEPTH() = CAST(" + (depth + 1) + " AS LONG) ) ";
+        }
+    }
+
+    protected String applyFindAllFilter(String query, String entityPath) {
+        String entityFilter = getFindAllFilter(entityPath);
+        if (entityFilter == null) {
+            return query;
+        } else {
+            String filter = " WHERE ";
+            if (query.toLowerCase().contains("where")) {
+                filter = " AND ";
+            }
+            filter += entityFilter;
+
+            query += filter;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "query". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1226)